### DISCUSSION TITLE
complete terraform & sam + cloudformation stack

### DISCUSSION THREAD

**Discussion by apps4u** [maplibre/martin#1523] [Ideas]
Hi All

I have created complete terraform stack that creates RDS postgres cluster with 1 or more db instances, it also has secrets rotation lambda and also a dbbootstrap lambda to enable postgres GIS extensions on the postgres RDS cluster. This also creates an s3 bucket to load the tile files if needed, it creates a cloud-front CDN and deploys Martin in a lambda.

I first created this a cloudformation+ sam to follow the doc but later ported the complete stack to terraform.

I have been given approval from work to donate this to the project if its something that is wanted if thats the case I will create documentation and do what ever else is required to sign this over.

Please

**Comment by nyurik** [comment]
We will be happy to add this to documentation. The biggest issue with this type of code is that it does not go into CI - so if something changes (e.g. in k8s, AWS, or anywhere else), the documentation will get stale.  Other than that - this would be awesome to have, especially in the Martin book section.

**Reply by apps4u** [reply]
cool I will document this and do PR on docs, 

I get it all good I will do this over next day or two. thanks
